qcodes/instrument_drivers/Shamrock/SR750.py

```diff
@@ -0,0 +1,305 @@
+        device_id (int): Spectrometer device ID.
+        serial_number (int): Serial number of the spectrometer.
+        number_gratings (int): Number of gratings on the spectrometer.
+    """
+
+    def __init__(self, name, dll_path=None, device_id=0, ccd_number_pixels=1024, ccd_pixel_width=26, **kwargs):
+
+        super().__init__(name, **kwargs)
+
+        # link to dll
+        self.ShamrockCIF = ShamrockCIF(dll_path=dll_path)
+
+        # store device number
+        self.device_id = device_id
+
+        # initialize Shamrock
+        self.ShamrockCIF.initialize()
+
+        # read info from Shamrock
+        self.serial_number = self.ShamrockCIF.get_serial_number(self.device_id)
+        self.number_gratings = self.ShamrockCIF.get_number_gratings(self.device_id)
+
+        # send CCD info to Shamrock
+        self.ShamrockCIF.set_number_pixels(self.device_id, ccd_number_pixels)
+        self.ShamrockCIF.set_pixel_width(self.device_id, ccd_pixel_width)
+
+        # add the instrument parameters
+        self.add_parameter('blaze',
+                           get_cmd=self.get_blaze,
+                           get_parser=int,
+                           label='blaze')
+
+        self.add_parameter('calibration',
+                           get_cmd=self.get_calibration,
+                           unit='nm',
+                           label='calibration')
+
+        self.add_parameter('grating',
+                           get_cmd=self.get_grating,
+                           set_cmd=self.set_grating,
+                           get_parser=int,
+                           vals=Ints(min_value=1,
+                                     max_value=self.number_gratings),
+                           label='grating')
+
+        self.add_parameter('groove_density',
+                           get_cmd=self.get_groove_density,
+                           get_parser=int,
+                           unit='l/mm',
+                           label='groove density')
+
+        self.add_parameter('slit',
+                           get_cmd=self.get_slit,
+                           set_cmd=self.set_slit,
+                           get_parser=int,
+                           vals=Ints(min_value=10,
+                                     max_value=2500),
+                           unit=u"\u03BC"+'m',
+                           label='slit')
+
+        min_wavelength, max_wavelength = self.ShamrockCIF.get_wavelength_limits(self.device_id, self.get_grating())
+        self.add_parameter('wavelength',
+                           get_cmd=self.get_wavelength,
+                           set_cmd=self.set_wavelength,
+                           get_parser=float,
+                           vals=Numbers(min_value=min_wavelength,
+                                        max_value=max_wavelength),
+                           unit='nm',
+                           label='wavelength')
+
+        # print connect message
+        self.connect_message(idn_param='IDN')
```

Comment:
This should be enough:
```suggestion
        self.connect_message()
```